core/src/main/java/nu/yona/server/subscriptions/service/UserAssertionService.java

```diff
@@ -68,29 +70,40 @@ static void assertValidUserFields(UserDto user, UserService.UserPurpose purpose)
 		}
 	}
 
-	public static void assertValidMobileNumber(String mobileNumber)
-	{
-		Require.that(REGEX_PHONE.matcher(mobileNumber).matches(), () -> InvalidDataException.invalidMobileNumber(mobileNumber));
-		assertNumberIsMobile(mobileNumber);
-	}
-
-	private static void assertNumberIsMobile(String mobileNumber)
+	public static void assertValidMobileNumber(String mobileNumberStr)
 	{
 		try
 		{
+			Require.that(REGEX_PHONE.matcher(mobileNumberStr).matches(),
+					() -> InvalidDataException.invalidMobileNumber(mobileNumberStr));
 			PhoneNumberUtil util = PhoneNumberUtil.getInstance();
-			PhoneNumberType numberType = util.getNumberType(util.parse(mobileNumber, null));
-			if ((numberType != PhoneNumberType.MOBILE) && (numberType != PhoneNumberType.FIXED_LINE_OR_MOBILE))
-			{
-				throw InvalidDataException.notAMobileNumber(mobileNumber, numberType.toString());
-			}
+			PhoneNumber mobileNumber = util.parse(mobileNumberStr, null);
+			assertNoLeadingZeros(util, mobileNumberStr, mobileNumber);
+			assertIsMobile(util, mobileNumberStr, mobileNumber);
 		}
 		catch (NumberParseException e)
 		{
 			throw YonaException.unexpected(e);
 		}
 	}
 
+	private static void assertNoLeadingZeros(PhoneNumberUtil util, String mobileNumberStr, PhoneNumber mobileNumber)
+	{
+		if (!mobileNumberStr.equals(util.format(mobileNumber, PhoneNumberFormat.INTERNATIONAL).replaceAll("[ -]", "")))
```

Comment:
This line is a bit hard to read for me. Does it really state that there are leading zeros? It seems that it is a very general validation...

Reply:
What it does is a "normalization" of the number: format the number in the international format and remove all decorations from it. Then the number should be identical to the original one.
Shall I add that in a comment?

Reply:
Added the comment